opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1954,7 +1954,9 @@ namespace Opm {
                 well_state.setCurrentProductionGroupControl(group.name(), newControl);
                 ss << "Switching control mode for group "<< group.name() << " to " << Group::ProductionCMode2String(newControl);
             }
-            wellGroupHelpers::setGroupControl(group, schedule(), reportStepIdx, false, well_state, ss);
+            // Pass a dummy phase for producer groups. The topPhase is only relevant for injector groups
+            const Phase topPhase = Phase::WATER;
+            wellGroupHelpers::setGroupControl(group, schedule(), topPhase, reportStepIdx, false, well_state, ss);
             break;
```

Comment:
Better to make a separate setInjectionGroupControl() and setProductionGroupControl() to avoid passing a dummy phase?